pallets/collator-selection/src/lib.rs

```diff
@@ -499,29 +512,83 @@ pub mod pallet {
 			collators.extend(candidates.into_iter().collect::<Vec<_>>());
 			collators
 		}
-		/// Kicks out and candidates that did not produce a block in the kick threshold.
+
+		/// Removes collators with unsatisfactory performance
+		/// Returns the removed AccountIds
 		pub fn kick_stale_candidates(
 			candidates: Vec<CandidateInfo<T::AccountId, BalanceOf<T>>>,
 		) -> Vec<T::AccountId> {
-			let now = frame_system::Pallet::<T>::block_number();
-			let kick_threshold = T::KickThreshold::get();
-			candidates
+			// 0. TODO: All sanity checks
+			let mut collator_perf_this_session =
+				<BlocksPerCollatorThisSession<T>>::iter().collect::<Vec<_>>();
+			if collator_perf_this_session.is_empty() {
+				return Vec::new();
+			}
+			// 1. Sort collator performance list
+			collator_perf_this_session.sort_unstable_by_key(|k| k.1); // XXX: don't like the tuple accessor, could this be a struct?
+														  // collator_perf_this_session.reverse();
+			let no_of_candidates = collator_perf_this_session.len();
+
+			// 2. get percentile by _exclusive_ nearest rank method https://en.wikipedia.org/wiki/Percentile#The_nearest-rank_method (rust percentile API is feature gated)
+			// RAD: Doublecheck that f64 as usize actually rounds up
```

Comment:
It actually truncates, so there needs to be a non_std equivalent of math::round::ceil here to satisfy https://en.wikipedia.org/wiki/Percentile#The_nearest-rank_method , any recommendations?

Reply:
resolved by using f64::ceil() from the FloatCore trait